packages/adblocker/src/engine/bucket/preprocessor.ts

```diff
@@ -0,0 +1,133 @@
+import { StaticDataView } from '../../data-view';
+import IFilter from '../../filters/interface';
+import Preprocessor, { Env } from '../../preprocessor';
+
+export default class PreprocessorBucket {
+  public static deserialize(view: StaticDataView): PreprocessorBucket {
+    const preprocessors: Preprocessor[] = [];
+
+    for (let i = 0, l = view.getUint32(); i < l; i++) {
+      preprocessors.push(Preprocessor.deserialize(view));
+    }
+
+    return new this({
+      preprocessors,
+    });
+  }
+
+  public readonly conditions: Map<string, Preprocessor>;
```

Comment:
This might change after the discussion about `generateDiff` but I think this could be a simple array of `Preprocessor`. The reason is that we only need the mapping from conditions to preprocessors as an optimization during the updates (which are very infrequent). If that's the case I would rather have the Map as an implementation detail of the update function (we can create it on the fly).

Reply:
I changed this to use `Array.prototype.find` instead of Map in the whole `Preprocessor` code. If it's not expected to have many entries and it's not expected to be called frequently, I feel reconstructing `Map` will slowdown the whole function.